nf_core/modules/lint/__init__.py

```diff
@@ -394,12 +394,16 @@ def update_meta_yml_file(self, mod):
                     log.debug(f"current ontologies for input: {current_ontologies_i}")
                     for ontology, ext in expected_ontologies_i:
                         if ontology not in current_ontologies_i:
-                            corrected_meta_yml["input"][i][j][element_name]["ontologies"].append(
-                                ruamel.yaml.comments.CommentedMap({"edam": ontology})
-                            )
-                            corrected_meta_yml["input"][i][j][element_name]["ontologies"][-1].yaml_add_eol_comment(
-                                f"{edam_formats[ext][1]}", "edam"
-                            )
+                            try:
+                                corrected_meta_yml["input"][i][j][element_name]["ontologies"].append(
+                                    ruamel.yaml.comments.CommentedMap({"edam": ontology})
+                                )
+                                corrected_meta_yml["input"][i][j][element_name]["ontologies"][-1].yaml_add_eol_comment(
+                                    f"{edam_formats[ext][1]}", "edam"
+                                )
+                            except Exception as e:
```

Comment:
I think it's better to just catch the specific exception. Could you double-check that the one you saw in your error was a `KeyError`? Thanks!
```suggestion
                            except KeyError as e:
```